### PR TITLE
feat(aiplatform): add sample for Gen AI code model tuning

### PR DESCRIPTION
**Pull request by gericdong**
## Description

Fixes b/293603484

Add a code sample to demonstrate how to create a code model tuning job.
Docs: https://cloud.google.com/vertex-ai/docs/generative-ai/models/tune-models#create_a_model_tuning_job

### CODE REVIEW

File: ai-platform/snippets/test/code-model-tuning.test.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    pipelineClient.cancelPipelineJob(cancelRequest).then(() => {
+      const deleteRequest = {
+        name,
+      };
+
+      return pipelineClient.deletePipeline(deleteRequest);
+    });
+  });
+  
+  beforeEach(stubConsole);
```
Comment (pattishin) [inline ai-platform/snippets/test/code-model-tuning.test.js:72]: **nit:** Just a  preference, normally `beforeEach` and `afterEach` are featured at the top of the test.
If nothing else is using `stubConsole` and `restoreConsole` functions - could you move them inline `beforeEach(function() { sinon.stub ...` .

File: ai-platform/snippets/test/code-model-tuning.test.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+'use strict';
+
+const {assert} = require('chai');
+const {describe, it} = require('mocha');
+const uuid = require('uuid')
+const sinon = require('sinon');
+
+const aiplatform = require('@google-cloud/aiplatform');
+const clientOptions = {
+  apiEndpoint: 'europe-west4-aiplatform.googleapis.com',
```
Comment (pattishin) [inline ai-platform/snippets/test/code-model-tuning.test.js:26]: **nit:** Swap hardcoded reference to `europe-west4` and add in `${location}`. Just so only one ref needs to change if this sample needs to be updated to another location.

File: ai-platform/snippets/code-model-tuning.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+      displayName: 'my-tuning-job',
+      runtimeConfig,
+    };
+
+    const createPipelineRequest = {
+      parent,
+      pipelineJob,
+      pipelineJobId,
+    };
+    await new Promise((resolve, reject) => {
```
Comment (pattishin) [inline ai-platform/snippets/code-model-tuning.js:79]: This outer promise can be removed and you can leverage the existing promise returned from line 80 and then chain it together like you've done on line 84.

```
pipelineClient.createPipelineJob(createPipelineRequest)
.then(...)
.then(...)
.catch(e => ...)
```

File: ai-platform/snippets/code-model-tuning.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  // const location = 'YOUR_PROJECT_LOCATION';
+  const aiplatform = require('@google-cloud/aiplatform');
+  const {PipelineServiceClient} = aiplatform.v1;
+
+  // Import the helper module for converting arbitrary protobuf.Value objects.
+  const {helpers} = aiplatform;
+
+  // Specifies the location of the api endpoint
+  const clientOptions = {
+    apiEndpoint: 'europe-west4-aiplatform.googleapis.com',
```
Comment (pattishin) [inline ai-platform/snippets/code-model-tuning.js:43]: **nit:** Replace `europe-west4` with potentially defined `location` variable.

**Review by pattishin — CHANGES_REQUESTED** [review]

File: ai-platform/snippets/test/code-model-tuning.test.js (hunk 1 above)
Comment (gericdong) [reply ai-platform/snippets/test/code-model-tuning.test.js]: Good to know. I am moving beforeEach and afterEach up in the test but keeping stubConsole and restoreConsole to be consistent with other samples. Thanks.

File: ai-platform/snippets/test/code-model-tuning.test.js (hunk 2 above)
Comment (gericdong) [reply ai-platform/snippets/test/code-model-tuning.test.js]: Changed. Thanks.

File: ai-platform/snippets/code-model-tuning.js (hunk 3 above)
Comment (gericdong) [reply ai-platform/snippets/code-model-tuning.js]: I removed the outer promise and changed the code to match the style of the generated samples.

**Review by pattishin — APPROVED** [review]
lgtm!

**Pull request merged**